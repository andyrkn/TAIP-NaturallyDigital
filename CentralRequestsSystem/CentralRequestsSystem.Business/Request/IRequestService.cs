using CentralRequestsSystem.Business.Request.Models;
using System.Threading.Tasks;

namespace CentralRequestsSystem.Business.Request
{
    public interface IRequestService
    {
        Task AddRequest(AddRequestModel addRequestModel);
    }
}
