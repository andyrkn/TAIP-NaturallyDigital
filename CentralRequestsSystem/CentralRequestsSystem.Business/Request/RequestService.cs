using System;
using System.Threading.Tasks;
using CentralRequestsSystem.Business.Request.Models;
using CentralRequestsSystem.Core;

namespace CentralRequestsSystem.Business.Request
{
    public class RequestService : IRequestService
    {
        private readonly IRequestRepository _requestRepository;
        public Task AddRequest(AddRequestModel addRequestModel)
            => throw new NotImplementedException();
    }
}
