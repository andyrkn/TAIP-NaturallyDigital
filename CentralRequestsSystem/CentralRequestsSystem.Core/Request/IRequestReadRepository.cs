using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestReadRepository : IReadRepository<Request>
    {
    }
}
