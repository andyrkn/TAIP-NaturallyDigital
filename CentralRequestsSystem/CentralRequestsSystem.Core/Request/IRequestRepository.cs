using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestRepository : IWriteRepository<Request>
    {
    }
}
