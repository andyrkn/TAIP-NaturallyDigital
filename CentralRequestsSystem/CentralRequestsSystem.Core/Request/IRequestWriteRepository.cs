using CentralRequestsSystem.Core.Repository;

namespace CentralRequestsSystem.Core
{
    public interface IRequestWriteRepository : IWriteRepository<Request>
    {
    }
}
