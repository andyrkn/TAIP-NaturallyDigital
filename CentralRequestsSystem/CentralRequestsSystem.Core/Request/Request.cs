using CSharpFunctionalExtensions;
using System;

namespace CentralRequestsSystem.Core
{
    public sealed class Request : Entity
    {
        private Request()
        {

        }

        public string UserAdress { get; private set; }
        public string IdentityProviderAdress { get; private set; }
        public DateTime Date { get; private set; }
        public string Payload { get; private set; }

        public static Result<Request> CreateRequest(string userAdress, string identityProviderAdress, DateTime date, string payload)
            => Result.Ok(new Request
                {
                    UserAdress = userAdress,
                    IdentityProviderAdress = identityProviderAdress,
                    Date = date,
                    Payload = payload
                });
    }
}
