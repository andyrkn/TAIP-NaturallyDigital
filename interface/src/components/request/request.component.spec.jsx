import React from "react";
import renderer from "react-test-renderer";
import Request from "./request.component";
import { assert } from "chai";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({
    adapter: new Adapter()
});

describe("Request component", () => {
    it("makes a snapshot", () => {
        const tree = renderer.create(<Request />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the institution and request type", () => {
        const wrapper = shallow(<Request institution="X" requestType="Y" />);
        const sections = wrapper.find("div");
        const institution = sections.getElements()[0];
        const requestType = sections.getElements()[1];

        assert.lengthOf(sections, 2);
        assert.equal(institution.props.children[1].props.children, "X");
        assert.equal(requestType.props.children[1].props.children, "Y");
    })
});
