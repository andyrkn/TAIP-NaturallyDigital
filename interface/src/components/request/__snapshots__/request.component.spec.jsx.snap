// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request component makes a snapshot 1`] = `
Array [
  <h2>
    Request
  </h2>,
  <div>
    <span
      className="key"
    >
      Institution: 
    </span>
    <span />
  </div>,
  <div>
    <span
      className="key"
    >
      Request type: 
    </span>
    <span />
  </div>,
]
`;
